### ISSUE TITLE
NullReferenceException crashes the application

### ISSUE THREAD

**Issue by raufismayilov** [TalAloni/SMBLibrary#257]
From time to time an application using Smb2Client crashs with the following exception:

```
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
at void SMBLibrary.Client.SMB2Client.OnClientSocketReceive(IAsyncResult ar)
at void System.Threading.Tasks.AwaitTaskContinuation.RunCallback(ContextCallback callback, object state, ref Task currentTask)
at void System.Threading.Tasks.Task.ThrowAsync(Exception exception, SynchronizationContext targetContext)+(object state) => { }
at void System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute()
at bool System.Threading.ThreadPoolWorkQueue.Dispatch()
at void System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()
```

That happens probably because buffer.AvailableLength tries to access internal m_buffer which is null because Dispose was called.

**Comment by raufismayilov** [comment]
This PR fixes this issue.
https://github.com/TalAloni/SMBLibrary/pull/258

**Comment by TalAloni** [comment]
I tried to analyze this issue previously: Getting NullReferenceException in OnClientSocketReceive when calling clientSocket.BeginReceive (in SMB2Client line 404) means that the connection buffer was disposed while receiving a packet.
I'm not sure how this can happen. if you can help me repro it will be very helpful.

**Comment by TalAloni** [comment]
I was able to reproduce this on the server implementation by invoking TerminateConnection.
I ended up using a lock to synchronize access to the NBTConnectionReceiveBuffer.
The fix was pushed to the repository.

**Comment by TalAloni** [comment]
This issue has been solved in v1.5.2
